PyInstaller/_shared_with_waf.py

```diff
@@ -50,6 +50,12 @@ def _pyi_machine(machine, system):
             return "arm"
         else:
             return "intel"
+        
+    if system == "SunOS":
+        if "sun" in machine.lower() or "sun" in platform.machine().lower():
```

Comment:
What exactly are `machine` and/or `platform.machine()` set to on `sparc`?

Reply:
I'm not sure about `machine` as it's passed in and I haven't seen where from. `platform.machine()` on SPARC depends on the processor but I've seen `sun4u` and `sun4v` hence checking for `sun`. On Solaris Intel it's always `i86pc` which is their version of i386/i686/x86_64

Reply:
If `i86pc` is always `i86pc` then just using `i86pc` would be fine. This whole name fuzzing only exists to deal with architectures with aliases or *short names* (`aarch64`/`arm64`/`armv8l`). If `DEST_CPU` on the build machine reliably matches `platform.machine()` on the runtime machine then it doesn't really matter what we call them.